warehouse/models/mart/gtfs/fct_monthly_shapes.sql

```diff
@@ -0,0 +1,70 @@
+{{ config(materialized='table') }}
+
+WITH dim_gtfs_datasets AS (
+    SELECT *
+    FROM {{ ref('dim_gtfs_datasets') }}
+),
+
+fct_scheduled_trips AS (
+    SELECT *
+    FROM {{ ref('fct_scheduled_trips') }}
+    WHERE service_date >= '2023-07-29' AND service_date <= '2023-07-31'
+),
+
+dim_shapes_arrays AS (
+
+    SELECT
+        key,
+        base64_url,
+        shape_id,
+        pt_array
+
+    FROM {{ ref('dim_shapes_arrays') }}
+    WHERE {{ incremental_where(default_start_var='GTFS_SCHEDULE_START',
+                               this_dt_column='_feed_valid_from',
+                               filter_dt_column='_feed_valid_from',
+                               dev_lookback_days = None)
+           }}
+),
+
+trips_by_shape AS (
+
+    SELECT
+        base64_url,
+        route_id,
+        shape_id,
+        EXTRACT(month FROM service_date) AS month,
+        EXTRACT(year FROM service_date) AS year,
+        COUNT(*) AS n_trips,
+
+    FROM fct_scheduled_trips
+    GROUP BY 1, 2, 3, 4, 5
+    QUALIFY ROW_NUMBER() OVER (PARTITION BY
+                               base64_url, route_id, month, year
+                               ORDER BY n_trips DESC) = 1
+),
+
+fct_monthly_shapes AS (
+
+    SELECT
+        dim_gtfs_datasets.source_record_id,
+        dim_gtfs_datasets.name,
+
+        trips.base64_url,
+        trips.route_id,
+        trips.shape_id,
+        trips.month,
+        trips.year,
+
+        shapes.key AS shape_array_key,
+        shapes.pt_array
+
+    FROM trips_by_shape AS trips
+    INNER JOIN dim_gtfs_datasets
+        ON trips.base64_url = dim_gtfs_datasets.base64_url
```

Comment:
This is going to result in fanout because `dim_gtfs_datasets` is versioned (i.e., there are a lot of rows with the same `base64_url`: one row per dataset record version). See line 41 for an alternate suggestion